### PR TITLE
feat: implemented discord oauth

### PR DESCRIPTION
**Pull request by BryanPan342**
Using discord oauth to see what user is logged in! I think this solution solves a lot of our issues.. Basically this is an easy way for us to see if the person has permissions because we check if they have the `board` role in our discord server.

https://user-images.githubusercontent.com/44623317/151718479-6e975b07-eb14-4cf2-8069-5fff42cfdc21.mov

### CODE REVIEW

**Review by tylercxu — APPROVED** [review]
lgtm nice idea

File: packages/web/website/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        },
+      });
+      if (!res.ok) {
+        return false;
+      }
+
+      const data: { roles: string[] } = await res.json();
+      const isBoardMember = (data?.roles ?? []).find(role => role === '761437487838789643');
+
+      return !!isBoardMember;
```
Comment (tylercxu) [inline packages/web/website/pages/api/auth/[...nextauth].ts:27]: isnt !! repetitive

**Review by tylercxu — CHANGES_REQUESTED** [review]

File: packages/web/website/pages/api/auth/[...nextauth].ts (hunk 1 above)
Comment (tylercxu) [reply packages/web/website/pages/api/auth/[...nextauth].ts:27]: wait nvm it just changes obj to bool

**Review by tylercxu — APPROVED** [review]
lgtm

**Comment by BryanPan342** [comment]
@mergifyio refresh

**Comment by BryanPan342** [comment]
@Mergifyio refresh

**Comment by BryanPan342** [comment]
@mergifyio refresh

**Pull request merged**